src/analyzer.rs

```diff
@@ -101,18 +101,37 @@ make_script! {
     Thai
 }
 
-pub struct AnalyzerConfig<'a, A> {
+pub struct AnalyzerConfig<'a, A = Vec<u8>> {
     /// language specialized tokenizer, this can be switched during
     /// document tokenization if the document contains several languages
     pub pipeline_map: HashMap<(Script, Language), Pipeline>,
-    pub stop_words: &'a Set<A>,
+    pub stop_words: Option<&'a Set<A>>,
 }
 
 impl<'a, A> AnalyzerConfig<'a, A>
 where
     A: AsRef<[u8]>,
 {
-    pub fn default_with_stopwords(stop_words: &'a Set<A>) -> Self {
+    pub fn with_stopwords(&mut self, stop_words: &'a Set<A>) -> &mut Self {
+        self.stop_words = Some(stop_words);
+        self
+    }
+
+    pub fn new(pipeline_map: HashMap<(Script, Language), Pipeline>, stop_words: Option<&'a Set<A>>) -> Self {
+        Self { pipeline_map, stop_words }
+    }
+}
+
+impl AnalyzerConfig<'_>
+{
+    pub fn new_with_pipeline(pipeline_map: HashMap<(Script, Language), Pipeline>) -> Self {
+        Self { pipeline_map, stop_words: None }
+    }
```

Comment:
This could be the `new` method of `AnalyzerConfig`, and if I want to add `stop_words` I'll call `with_stopwords(...)` method

Reply:
Done thanks :+1: